src/mainwindow.cpp

```diff
@@ -250,45 +225,49 @@ void MainWindow::openProject() {
         return;
     }
 
-    _loadProject(projectDirectory);
+    loadProject(projectDirectory);
 
 }
 
-void MainWindow::_loadProject(const QString& projectDirectory) {
+void MainWindow::loadProject(const QString& projectDirectory) {
 
-    _connectScenes();
+    connectScenes();
 
     m_currentProject = std::make_shared<Editor::Project>(
         std::filesystem::path(projectDirectory.toStdString()).string()
     );
 
-    ui->treeViewMaps->setModel(
+    m_ui->treeViewMaps->setModel(
         static_cast<QAbstractItemModel*>(m_currentProject->mapsModel())
     );
 
-    ui->treeViewMaps->setProject(m_currentProject);
+    m_ui->treeViewMaps->setProject(m_currentProject);
 
     // Enable the first layer drawing by default.
-    ui->actionLow_layer_1->trigger();
+    m_ui->actionLow_layer_1->trigger();
 }
 
 void MainWindow::saveProject() {
-    if (nullptr != m_currentProject) {
-        m_currentProject->saveProjectFile();
-        qDebug() << m_currentProject->openedMaps().count();
-        if (m_currentProject->openedMaps().count() > 0) {
-            QMap<QString, Misc::MapDocument>::iterator i;
-
-            for(auto e : m_currentProject->openedMaps().keys()) {
-                qDebug() << e;
-                m_currentProject->document(e)->save();
-            }
-        }
```

Comment:
Here there is two approachs : 
- no early return (your choice) 
- avoid too much nesting if this can be avoided, even if the cost is some early returns.

What I recommand : 
When check test is "test if conditions are ok, then continue", it's better to think as "if the test is not ok, stop here".
So this give a better readability and less tabulations and nesting